tests/ui/redundant_test_prefix.rs

```diff
@@ -0,0 +1,11 @@
+#![warn(clippy::redundant_test_prefix)]
```

Comment:
Could you add more tests? Also negative examples, where the `test_` prefix should be kept. For example, in normal functions and methods.

Reply:
The normal functions and methods example makes sense, but can you (or anyone else here) think of other examples?

Reply:
I think those would be enough. One specific example I can think of would be:

```
mod no_attr {
    pub mod tests {
       pub fn test_nothing() {}
    }
}
```

This looks very close to unit tests, but is missing the attributes. This should not be linted.